modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/GoogleSpeechStartTranscriptionOperationHandler.java

```diff
@@ -0,0 +1,185 @@
+   * The configuration options for this handler
+   */
+  private static final SortedMap<String, String> CONFIG_OPTIONS;
+
+  static {
+    CONFIG_OPTIONS = new TreeMap<String, String>();
+    CONFIG_OPTIONS.put(SOURCE_FLAVOR, "The \"flavor\" of the track to use as audio input");
+    CONFIG_OPTIONS.put(SOURCE_TAG, "The \"tag\" of the track to use as audio input");
+    CONFIG_OPTIONS.put(LANGUAGE_CODE, "The \"language code\" to use for the transcription");
+    CONFIG_OPTIONS
+            .put(SKIP_IF_FLAVOR_EXISTS, "If this \"flavor\" is already in the media package, skip this operation");
+  }
+
+  @Override
+  protected void activate(ComponentContext cc) {
+    super.activate(cc);
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see
+   * org.opencastproject.workflow.api.WorkflowOperationHandler#start(org.opencastproject.workflow.api.WorkflowInstance,
+   * JobContext)
+   */
+  @Override
+  public WorkflowOperationResult start(final WorkflowInstance workflowInstance, JobContext context)
+          throws WorkflowOperationException {
+    MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+
+    String skipOption = StringUtils.trimToNull(operation.getConfiguration(SKIP_IF_FLAVOR_EXISTS));
+    if (skipOption != null) {
+      MediaPackageElement[] mpes = mediaPackage.getElementsByFlavor(MediaPackageElementFlavor.parseFlavor(skipOption));
+      if (mpes != null && mpes.length > 0) {
+        logger.info(
+                "Start transcription operation will be skipped because flavor {} already exists in the media package",
+                skipOption);
+        return createResult(Action.SKIP);
+      }
+    }
+
+    logger.debug("Start transcription for mediapackage {} started", mediaPackage);
+
+    // Get language code if configured
+    String langCode = StringUtils.trimToNull(operation.getConfiguration(LANGUAGE_CODE));
+
+    // Check which tags have been configured
+    String sourceTagOption = StringUtils.trimToNull(operation.getConfiguration(SOURCE_TAG));
+    String sourceFlavorOption = StringUtils.trimToNull(operation.getConfiguration(SOURCE_FLAVOR));
+
+    AbstractMediaPackageElementSelector<Track> elementSelector = new TrackSelector();
+
+    // Make sure either one of tags or flavors are provided
+    if (StringUtils.isBlank(sourceTagOption) && StringUtils.isBlank(sourceFlavorOption)) {
+      throw new WorkflowOperationException("No source tag or flavor have been specified!");
+    }
+
+    if (StringUtils.isNotBlank(sourceFlavorOption)) {
+      String flavor = StringUtils.trim(sourceFlavorOption);
+      try {
+        elementSelector.addFlavor(MediaPackageElementFlavor.parseFlavor(flavor));
+      } catch (IllegalArgumentException e) {
+        throw new WorkflowOperationException("Source flavor '" + flavor + "' is malformed");
+      }
+    }
+    if (StringUtils.isNotBlank(langCode)) {
+      language = StringUtils.trim(langCode);
```

Comment:
`langCode` has already been trimmed to null above.  IMO line 120 should read `language = ...` since `language` as implemented ends up as either the non-blank `langCode` value, or `null`.

Reply:
fixe